libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -1821,13 +1821,14 @@ AP_AHRS::EKFType AP_AHRS::active_EKF_type(void) const
         if (hal.util->get_soft_armed() &&
             (!filt_state.flags.using_gps ||
              !filt_state.flags.horiz_pos_abs) &&
+             !filt_state.flags.horiz_pos_rel &&
```

Comment:
this is a tricky one. It means that we are trusting the EKF to judge if it can provide a horizontal position. When flying in an area where jamming may be expected then this is obviously a good change. If flying in an area where jamming is not expected then I think it is far more likely that the EKF is wrong and the GPS is right (which is why this whole fallback to DCM logic is there). The EKF does sometimes go back, either with vibration or bad yaw from GSF (especially if no compass). 
Maybe we need an AHRS_OPTIONS bit for when we are flying in an area where jamming is expected?